src/CoreWCF.Primitives/src/CoreWCF/AuthorizeRoleAttribute.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using CoreWCF.Description;
+using CoreWCF.Dispatcher;
+using CoreWCF.IdentityModel.Claims;
+
+namespace CoreWCF
+{
+    [AttributeUsage(CoreWCFAttributeTargets.OperationContract, Inherited = true)]
+    public class AuthorizeRoleAttribute : Attribute, IAuthorizeOperation
+    {
+        public string AllowedRoles { get; set; }
+
```

Comment:
I'm not 100% on the exact api shape here. It seems odd to have a property and not a constructor. Generally attribute properties are for optional value and there's normally a constructor or two for common scenarios. Only having a property seems to be going against convention. Having a constructor with a params argument would make the multi-role scenario simpler. So if you did this:
```c#
private string[] _allowedRoles;
public AuthorizeRoleAttribute(params string[] allowedRoles)
{
    _allowedRoles = allowedRoles;
}
```
Then in BuildClaim you can iterate on the array. Usage would be:
```c#
        [AuthorizeRole("InvalidRole", "CoreWCFGroupAdmin")]
        public string EchoForAuthorizarionOneRole(string value) { return value; }
```

Reply:
I like your idea... it's easy.. will shape accordingly